source/bringauto/modules/devices/testing_device/testing_module_manager.cpp

```diff
@@ -0,0 +1,62 @@
+#include <fleet_protocol/module_maintainer/module_gateway/module_manager.h>
+
+#include <cstring>
+#include <iostream>
+#include <string>
+
+// bringauto::modules::transparent_module::testing_module_manager
+namespace bringauto::modules::transparent_module::devices::testing_device
+{
+
+    int testing_device_send_status_condition(const buffer current_status, const buffer new_status) { return OK; }
+
+    int testing_device_generate_command(buffer *generated_command, const buffer new_status, const buffer current_status,
+                                        const buffer current_command)
+    {
+        if (allocate(generated_command, current_command.size_in_bytes) == NOT_OK)
+        {
+            return NOT_OK;
+        }
+        std::memcpy(generated_command->data, current_command.data, generated_command->size_in_bytes);
+
+        return OK;
+    }
```

Comment:
This improvement is not necessary because the command only depends on the current_command generated by the device (in our case, likely the user during testing). The function works the same as a similar function in the io-module for button implementation.